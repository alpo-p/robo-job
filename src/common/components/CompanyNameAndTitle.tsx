import { useTheme } from '@react-navigation/native'
import React from 'react'
import { View } from 'react-native'
import styles from '../styles'
import RoboText from './RoboText'

interface P {
  companyName: string
  jobTitle: string
  editStylesForDetailedView?: boolean
}
export default ({ companyName, jobTitle, editStylesForDetailedView }: P) => {
  const { colors } = useTheme()
  const b = editStylesForDetailedView

  return (
    <View
      style={{
        alignItems: b ? undefined : 'center',
      }}
    >
      <RoboText
        uppercase={!b}
        color={b ? colors.primary : styles.white}
        size="large"
        weight="bold"
      >
        {companyName}
      </RoboText>
      {editStylesForDetailedView && (
        <View
          style={{
            marginTop: 4,
          }}
        />
      )}
      <RoboText color={b ? colors.text : styles.white} size="large">
        {jobTitle}
      </RoboText>
    </View>
  )
}
