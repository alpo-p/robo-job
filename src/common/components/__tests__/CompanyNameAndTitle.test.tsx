import React from 'react'
import renderer from 'react-test-renderer'
import CompanyNameAndTitle from '../CompanyNameAndTitle'

describe('CompanyLogo', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <CompanyNameAndTitle companyName="test name" jobTitle="test title" />,
    )
    expect(tree).toMatchSnapshot()
  })
})
