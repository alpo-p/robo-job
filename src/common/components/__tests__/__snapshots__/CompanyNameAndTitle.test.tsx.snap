// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyLogo should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "DamascusBold",
        "fontSize": 20,
        "textTransform": "uppercase",
      }
    }
  >
    test name
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Damascus",
        "fontSize": 20,
        "textTransform": "none",
      }
    }
  >
    test title
  </Text>
</View>
`;
